bot/handlers/not_sending_videos/save_clip_handler.py

```diff
@@ -35,206 +32,95 @@
+            return output_filename.replace(" ", "_"), 0.0, 0.0, is_compilation, None, None
 
-        if "episode_info" in segment_info_dict and isinstance(segment_info_dict["episode_info"], dict):
-            episode_info_data = segment_info_dict["episode_info"]
-            segment_info_dict["episode_info"] = EpisodeInfo(
-                season=episode_info_data.get("season"),
-                episode_number=episode_info_data.get("episode_number"),
+        if last_clip.clip_type == ClipType.ADJUSTED:
+            await ClipsExtractor.extract_clip(
+                segment_dict.get("video_path"), last_clip.adjusted_start_time, last_clip.adjusted_end_time,
+                output_filename, self._logger,
             )
+            return output_filename, last_clip.adjusted_start_time, last_clip.adjusted_end_time, False, season, episode_number
 
-    async def __validate_required_fields(self, segment_info_dict: Dict) -> bool:
-        required_fields = ["video_path", "start", "end", "episode_info"]
-        missing_fields = [field for field in required_fields if field not in segment_info_dict]
-        if missing_fields:
-            await self._log_system_message(logging.ERROR, f"Missing required fields for SegmentInfo: {segment_info_dict}")
-            return False
-        return True
-
-    @staticmethod
-    def __create_segment_info(segment_info_dict: Dict) -> SegmentInfo:
-        return SegmentInfo(
-            video_path=segment_info_dict["video_path"],
-            start=segment_info_dict["start"],
-            end=segment_info_dict["end"],
-            episode_info=segment_info_dict["episode_info"],
-            text=segment_info_dict.get("text"),
-            id=segment_info_dict.get("id"),
-            author=segment_info_dict.get("author"),
-            comment=segment_info_dict.get("comment"),
-            tags=segment_info_dict.get("tags"),
-            location=segment_info_dict.get("location"),
-            actors=segment_info_dict.get("actors"),
-            compiled_clip=segment_info_dict.get("compiled_clip"),
-        )
-
-    @staticmethod
-    def _convert_compiled_to_segment_info(last_clip_info: LastClip) -> SegmentInfo:
-        return SegmentInfo(
-            video_path="default_path.mp4",
-            start=0.0,
-            end=last_clip_info.adjusted_end_time or 0.0,
-            episode_info=EpisodeInfo(season=None, episode_number=None),
-            compiled_clip=last_clip_info.compiled_clip,
-        )
-
-    @staticmethod
-    def _convert_to_segment_info(segment: json) -> SegmentInfo:
-        episode_info_data = segment.get("episode_info")
-
-        if isinstance(episode_info_data, EpisodeInfo):
-            episode_info_obj = episode_info_data
-        else:
-            episode_info_obj = EpisodeInfo(
-                season=episode_info_data["season"],
-                episode_number=episode_info_data["episode_number"],
+        if last_clip.clip_type == ClipType.MANUAL:
```

Comment:
nie no czemu te lambdy wywaliles, lepiej bylo a teraz masz nawalone 6 ifow chyba

Reply:
mówisz to mi się wydało prostsze a te lambdy mniej rozumiałem ale można wrócić

Reply:
mozna a nawet trzeba